app/domain/authentication/authn_k8s/inject_client_cert.rb

```diff
@@ -50,6 +51,22 @@ def install_signed_cert
         validate_cert_installation(resp)
       end
 
+      # In the old version of the authn-client we assumed that the host is under the "apps" policy branch.
+      # Now we send the host-id in 2 parts:
+      #   suffix - the last 3 parts that indicate the machine identity (e.g ${TEST_APP_NAMESPACE_NAME}/*/*)
+      #   prefix - all the rest (e.g host/conjur/authn-k8s/${AUTHENTICATOR_ID})
+      # We update the CSR's common_name to have the full host-id. This way, the validation
+      # that happens in the "authenticate" request will work, as the signed certificate
+      # contains the full host-id.
+      def update_csr_common_name
```

Comment:
Is it possible for the `smart_csr` to generate the correct common name for itself?  It feels like the abstraction is leaking a bit.

Reply:
I don't understand your suggestion. 

The logic here is:
1. understand wether the client sent a custom prefix
2. Deciding which is the corresponding prefix according to (1)
3. Update the common-name to have the prefix chosen in (2)

Are you suggesting to move this logic into `SmartCsr`?